app/src/main/java/org/gem/indo/dooit/api/managers/DooitManager.java

```diff
@@ -80,8 +87,30 @@ public Response intercept(Chain chain) throws IOException {
             }
         });
         httpClient.addInterceptor(logging);
-
-        OkHttpClient client = httpClient.build();
+        if(doOfflineCache) {
+            final Cache cache = new Cache(application.getCacheDir(), 10 * 1024 * 1024);
+            this.client = OkCacheControl.on(httpClient)
+                    .overrideServerCachePolicy(30, MINUTES)
+                    .forceCacheWhenOffline(new OkCacheControl.NetworkMonitor() {
+                        @Override
+                        public boolean isOnline() {
+                            ConnectivityManager cm =
+                                    (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+                            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+                            boolean isConnected = activeNetwork != null &&
+                                    activeNetwork.isConnectedOrConnecting();
+
+                            return isConnected;
+                        }
+                    })
+                    .apply() // return to the OkHttpClient.Builder instance
+                    .cache(cache)
+                    .build();
```

Comment:
We need some way to flush specific URLs from the cache at will, for example retrieving goals from the server, after a goal has been added the retrieving goals endpoint needs to run if the URL is in the cache it wont retrieve the last goal you added